src/pip/_internal/cli/req_command.py

```diff
@@ -231,6 +231,7 @@ def make_requirement_preparer(
             finder=finder,
             require_hashes=options.require_hashes,
             use_user_site=use_user_site,
+            use_shallow_wheels=getattr(options, 'shallow_wheels', False),
```

Comment:
I think this should just become `options.shallow_wheels`, but it's not clear to me whether we want to add this `--shallow-wheels` argument like this, or whether it should be some separate global option.